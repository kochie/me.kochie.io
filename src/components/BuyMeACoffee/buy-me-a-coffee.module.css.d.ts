export const bmcButton: string
export const bmcSpan: string
export const bmcButtonContainer: string
