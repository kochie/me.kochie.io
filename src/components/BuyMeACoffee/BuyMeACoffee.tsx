import * as React from 'react'

import './BuyMeACoffee.css'

interface Props {
  username: String
}

const coffee = "https://www.buymeacoffee.com/assets/img/BMC-btn-logo.svg"

const BuyMeACoffee: React.SFC<Props> = ({username}) => {
  return (
    <div className="bmc-button-container">
      <a className="bmc-button" target="_blank" href={`https://www.buymeacoffee.com/${username}`}>
        <img src={coffee} alt="Buy me a coffee" />
        <span className="bmc-span">Buy me a coffee</span>
      </a>
    </div>
  )
}

export default BuyMeACoffee