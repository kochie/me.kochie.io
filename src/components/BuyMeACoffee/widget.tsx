import React, { ReactElement } from 'react'
import Head from 'next/head'

export default function Widget(): ReactElement {
  return (
    <Head>
      <script
        data-name="BMC-Widget"
        src="https://cdnjs.buymeacoffee.com/1.0.0/widget.prod.min.js"
        data-id="kochie"
        data-description="Support me on Buy me a coffee!"
        data-message="Thank you for visiting. You can now buy me a coffee!"
        data-color="#FF813F"
        data-position="right"
        data-x_margin="18"
        data-y_margin="18"
      />
    </Head>
  )
}
