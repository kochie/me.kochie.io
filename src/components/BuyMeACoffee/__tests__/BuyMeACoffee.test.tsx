import * as renderer from 'react-test-renderer'
import * as React from 'react'

import BuyMeACoffee from '../BuyMeACoffee'

it('by me a coffee renders correctly', () => {
  const tree = renderer.create(<BuyMeACoffee username="test" />).toJSON()
  expect(tree).toMatchSnapshot()
})
