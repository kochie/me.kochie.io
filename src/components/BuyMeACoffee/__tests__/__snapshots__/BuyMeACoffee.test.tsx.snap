// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`by me a coffee renders correctly 1`] = `
<div
  className="bmcButtonContainer"
>
  <a
    className="bmcButton"
    href="https://www.buymeacoffee.com/test"
    rel="noopener"
    target="_blank"
  >
    <img
      alt="Buy me a coffee"
      src="https://www.buymeacoffee.com/assets/img/BMC-btn-logo.svg"
    />
    <span
      className="bmcSpan"
    >
      Buy me a coffee
    </span>
  </a>
</div>
`;
